code401challenges/linkedLists/src/main/java/linkedLists/LinkedList.java

```diff
@@ -1,7 +1,34 @@
 package linkedLists;
 
-public class LinkedList {
-    Node head; // head of list
+public class LinkedList<T> {
+    Node<T> head; // head of list
+
+    public LinkedList() {
+        this.head = null;
+    }
+
+    // Method to insert a new node
+    public LinkedList Insert(T value) {
+        LinkedList list = new LinkedList();

```

Comment:
You don't need to create and return a new linked list. If you follow this method line by line, this new LinkedList is actually empty. This method doesn't actually need to return anything!